cmd/fortifyExecuteScan.go

```diff
@@ -822,3 +886,39 @@ func getSeparator() string {
 	}
 	return ":"
 }
+
+func ensureAuthEntitiesExist(projectVersionId int64, sys fortify.System, config *fortifyExecuteScanOptions) {
+	entityList, err := sys.GetAuthEntityOfProjectVersion(projectVersionId)
+	if err != nil {
+		log.Entry().WithError(err).Fatalf("Failed to fetch auth entities for project version %v", projectVersionId)
+	}
+
+	// For each ensureId in config.AuthEntityIDs, ensure that it exists in the current project version's collection of auth entities.
+	for _, ensureIDStr := range strings.Split(config.AuthEntityIDs, ",") {
+		found := false
+		ensureIDStr = strings.TrimSpace(ensureIDStr)
+		ensureID, err := strconv.Atoi(ensureIDStr)
+
+		if err != nil {
+			log.Entry().WithError(err).Fatalf("Failed to convert entity ID to int: %s", ensureIDStr)
+		}
+
+		for _, entity := range entityList {
+			if entity.ID == int64(ensureID) {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			newEntity := &models.AuthenticationEntity{
+				ID:     int64(ensureID),
+				IsLdap: true,
+			}
+			err := sys.UpdateCollectionAuthEntityOfProjectVersion(projectVersionId, []*models.AuthenticationEntity{newEntity})
```

Comment:
The signature (generated by swagger?) is very strange. You would expect that you have to pass the complete list of entities and the function replaces the list instead of adds it to the existing list.